src/jit/importcall.cpp

```diff
@@ -0,0 +1,1714 @@
+
+            if (clsFlags & CORINFO_FLG_ARRAY)
+            {
+                compiler->eeGetCallSiteSig(resolvedToken->token, resolvedToken->tokenScope, resolvedToken->tokenContext,
+                                           sig);
+            }
+
+            // Find the return type used for verification by interpreting the method signature.
+            // NB: we are clobbering the already established sig.
+            if (compiler->tiVerificationNeeded)
+            {
+                // Actually, we never get the sig for the original method.
+                sig = &(callInfo->verSig);
+            }
+
+            typeInfo tiRetVal = compiler->verMakeTypeInfo(sig->retType, sig->retTypeClass);
+            tiRetVal.NormaliseForStack();
+
+            // The CEE_READONLY prefix modifies the verification semantics of an Address
+            // operation on an array type.
+            if ((clsFlags & CORINFO_FLG_ARRAY) && readonlyCall && tiRetVal.IsByRef())
+            {
+                tiRetVal.SetIsReadonlyByRef();
+            }
+
+            if (compiler->tiVerificationNeeded)
+            {
+                // We assume all calls return permanent home byrefs. If they
+                // didn't they wouldn't be verifiable. This is also covering
+                // the Address() helper for multidimensional arrays.
+                if (tiRetVal.IsByRef())
+                {
+                    tiRetVal.SetIsPermanentHomeByRef();
+                }
+            }
+
+            if (call->IsCall())
+            {
+                // Sometimes "call" is not a GT_CALL (if we imported an intrinsic that didn't turn into a call)
+
+                bool fatPointerCandidate = call->AsCall()->IsFatPointerCandidate();
+                if (varTypeIsStruct(callRetTyp))
+                {
+                    call = compiler->impFixupCallStructReturn(call->AsCall(), sig->retTypeClass);
+                }
+
+                if ((call->gtFlags & GTF_CALL_INLINE_CANDIDATE) != 0)
+                {
+                    assert(compiler->opts.OptEnabled(CLFLG_INLINING));
+                    assert(!fatPointerCandidate); // We should not try to inline calli.
+
+                    // Make the call its own tree (spill the stack if needed).
+                    compiler->impAppendTree(call, (unsigned)compiler->CHECK_SPILL_ALL, compiler->impCurStmtOffs);
+
+                    // TODO: Still using the widened type.
+                    call = compiler->gtNewInlineCandidateReturnExpr(call, genActualType(callRetTyp));
+                }
+                else
+                {
+                    if (fatPointerCandidate)
+                    {
+                        // fatPointer candidates should be in statements of the form call() or var = call().
+                        // Such form allows to find statements with fat calls without walking through whole trees
+                        // and removes problems with
```

Comment:
It would be useful if these member variable declarations (eventually) had comments. I realize you copy-and-pasted them, so no need to do it now.

Are they all required to be member vars? Are all of them used by at least 2 different functions?